### PR TITLE
Parser: implement import

### PR DESCRIPTION
**Pull request by kazcw**
### Pull Request Description

Based on usage; I believe this handles every case in current `.enso` files.

### Important Notes

- `import` is a built-in macro, so an import statement parses as a `MultiSegmentApp`.
- Every `import` syntax will have a segment whose leading keyword is `import`; however `import` macros can be identified more efficiently by looking at only the first keyword. A `MultiSegmentApp` is an import if and only if its first keyword is in the set { "polyglot", "from", "import" }.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide dist` and `./run ide watch`.

[ci no changelog needed]

### CODE REVIEW

File: lib/rust/parser/tests/parse.rs [hunk 1]
```diff
@@ -362,6 +362,49 @@ fn plus_negative() {
...
 
 
+// === Import ===
+
+#[test]
+fn import() {
+    #[rustfmt::skip]
+    let cases = [
+        ("import project.IO", block![
+            (MultiSegmentApp #(((Ident import) (OprApp (Ident project) (Ok ".") (Ident IO)))))]),
```
Comment (JaroslavTulach) [inline lib/rust/parser/tests/parse.rs]: Why do you believe an `import` shall parse as `MultiSegmentApp`?

**Review by JaroslavTulach — CHANGES_REQUESTED** [review]
Please create `Tree.Import` and capture the import info in there.

**Comment by kazcw** [comment]
We are implementing this sort of thing with "macros" (MultiSegmentApp)

**Comment by JaroslavTulach** [comment]
If I properly remember our yesterday conversation, then `MultiSegmentApp` is suitable representation for `if/then/else`, but `import` would benefit from its own `Tree` structure. The less logic I can do in [TreeToIr](https://github.com/enso-org/enso/pull/3611/files#diff-ba87b05069ce6d6bc2bcc063f40bf85389597c107af7f20352604b701eb7ad86) the better.

File: lib/rust/parser/tests/parse.rs [hunk 2]
```diff
@@ -151,6 +148,22 @@ fn type_def_full() {
...
 
+#[test]
+fn type_def_nested() {
+    let code = [
+        "type Foo",
+        "    type Bar",
+        "    type Baz",
+    ];
+    #[rustfmt::skip]
+    let expected = block![
```
Comment (JaroslavTulach) [inline lib/rust/parser/tests/parse.rs:160]: +1, thanks for this test.

**Review by JaroslavTulach — APPROVED** [review]
Looks fine. Chatting with Kaz over the table revealed that `import` is a Java keyword and we need to prefix the field with something. Otherwise OK.

File: lib/rust/metamodel/src/java/from_meta.rs [hunk 3]
```diff
@@ -106,7 +106,9 @@ impl FromMeta {
         let mut fields = Vec::with_capacity(fields_.size_hint().0);
         for field in fields_ {
             let meta::Field { name, type_, hide, .. } = field;
-            let name = name.to_camel_case().expect("Unimplemented: Tuples.");
+            let mut name_ = meta::FieldName::from_snake_case("field");
```
Comment (JaroslavTulach) [inline lib/rust/metamodel/src/java/from_meta.rs:109]: This change renamed even the public getters. Now I see `Line.getFieldExpression()` methods & co. Workarounded by https://github.com/enso-org/enso/pull/3611/commits/aa2b63a1863a0c775e703620d71213cb8c2c73c9#r940901973

**Pull request merged**